querydsl-sql/src/main/java/com/querydsl/sql/Union.java

```diff
@@ -25,23 +25,17 @@
  *
  * @param <RT> return type of projection
  */
-public interface Union<RT> extends SubQueryExpression<RT> {
+public interface Union<RT> extends SubQueryExpression<RT>, Fetchable<RT> {
 
     /**
      * Get the projection as a typed List
      *
-     * @return results
+     * @deprecated Use {@link Union#fetch()}
      */
+    @Deprecated
     List<RT> list();
 
     /**
-     * Get the projection as a typed Iterator
-     *
-     * @return result iterator
-     */
-    CloseableIterator<RT> iterate();
```

Comment:
I would keep this method here, because I am not sure if it is binary compatible.
It also doesn't hurt that much.